pallets/payment-streams/src/utils.rs

```diff
@@ -0,0 +1,490 @@
+        ensure!(
+            !UsersWithoutFunds::<T>::contains_key(user_account),
+            Error::<T>::UserWithoutFunds
+        );
+
+        // Remove the payment stream from the PaymentStreams mapping
+        PaymentStreams::<T>::remove(bsp_id, user_account);
+
+        // Decrease the user's payment streams count
+        let mut user_payment_streams_count = RegisteredUsers::<T>::get(user_account);
+        user_payment_streams_count = user_payment_streams_count
+            .checked_sub(1)
+            .ok_or(ArithmeticError::Underflow)?;
+        RegisteredUsers::<T>::insert(user_account, user_payment_streams_count);
+
+        // If the user has no more payment streams, release the deposit
+        if user_payment_streams_count == 0 {
+            // Release the deposit from the user
+            T::NativeBalance::release_all(
+                &HoldReason::PaymentStreamStorageDeposit.into(),
+                &user_account,
+                Precision::Exact,
+            )?;
+
+            // Remove the user from the UsersWithoutFunds mapping
+            UsersWithoutFunds::<T>::remove(user_account);
+        }
+
+        Ok(())
+    }
+
+    /// This function holds the logic that checks if a payment stream can be charged and, if so, charges the payment stream from the user's balance.
+    /// The charge is calculated as: rate * time_passed where time_passed is the time between the last valid proof submitted and the last charged proof of this payment stream.
+    /// As such, the last charged proof can't be greater than the last valid proof, and if they are equal then no charge is made.
+    ///
+    /// Note: right now, the other functions receive the BSP's account ID because it is needed to transfer the user payment to it in this function, but the idea is to change the
+    /// BSP struct to hold a `payment_address` where the payment should be received, and then the functions of this pallet will receive the BSP's ID instead.
+    pub fn do_charge_payment_stream(
+        bsp_account: &T::AccountId,
+        user_account: &T::AccountId,
+    ) -> Result<BalanceOf<T>, DispatchError> {
+        // Get the BSP ID of the BSP account
+        let bsp_id = <T::Providers as storage_hub_traits::ProvidersInterface>::get_provider(
+            bsp_account.clone(),
+        )
+        .ok_or(Error::<T>::NotABackupStorageProvider)?;
+
+        // Check that a payment stream between that BSP and user exists
+        ensure!(
+            PaymentStreams::<T>::contains_key(bsp_id, user_account),
+            Error::<T>::PaymentStreamNotFound
+        );
+
+        // Get the information of the payment stream
+        let payment_stream = PaymentStreams::<T>::get(bsp_id, user_account)
+            .ok_or(Error::<T>::PaymentStreamNotFound)?;
+
+        // Note: No need to check if a new proof has been submitted since the last charge as the only consecuence of that is charging 0 to the user,
```

Comment:
```suggestion
        // Note: No need to check if a new proof has been submitted since the last charge as the only consequence of that is charging 0 to the user,
```